test/Npgsql.Tests/SchemaTests.cs

```diff
@@ -145,6 +146,95 @@ public void DataSourceInformation()
 
+        IEnumerable<int> Foo()
+        {
+            try
+            {
+                for (var i = 0; i < 5; i++)
+                    yield return i;
+            }
+            finally
+            {
+                Console.WriteLine("IN FINALLY");
+            }
+            for (var i = 5; i < 10; i++)
+                yield return i;
+        }
+
+        [Test]
+        public void DataTypes()
+        {
+            using (var conn = OpenConnection())
+            {
+                conn.ExecuteNonQuery("CREATE TYPE pg_temp.test_enum AS ENUM ('a', 'b')");
+                conn.ExecuteNonQuery("CREATE TYPE pg_temp.test_composite AS (a INTEGER)");
+                conn.ExecuteNonQuery("CREATE DOMAIN pg_temp.us_postal_code AS TEXT");
+                conn.ReloadTypes();
+                conn.TypeMapper.MapEnum<TestEnum>();
+                conn.TypeMapper.MapComposite<TestComposite>();
+
+                var metadata = conn.GetSchema(DbMetaDataCollectionNames.MetaDataCollections).Rows
+                    .Cast<DataRow>()
+                    .Single(r => (string)r["CollectionName"] == "DataTypes");
+                Assert.That(metadata["NumberOfRestrictions"], Is.Zero);
+                Assert.That(metadata["NumberOfIdentifierParts"], Is.Zero);
+
+                var dataTypes = conn.GetSchema(DbMetaDataCollectionNames.DataTypes);
+
+                var intRow = dataTypes.Rows.Cast<DataRow>().Single(r => (string)r["TypeName"] == "integer");
+                Assert.That(intRow["DataType"], Is.EqualTo("Int32"));
+                Assert.That(intRow["ProviderDbType"], Is.EqualTo((int)NpgsqlDbType.Integer));
+                Assert.That(intRow["IsUnsigned"], Is.False);
+                Assert.That(intRow["OID"], Is.EqualTo(23));
+
+                var textRow = dataTypes.Rows.Cast<DataRow>().Single(r => (string)r["TypeName"] == "text");
+                Assert.That(textRow["DataType"], Is.EqualTo("String"));
+                Assert.That(textRow["ProviderDbType"], Is.EqualTo((int)NpgsqlDbType.Text));
+                Assert.That(textRow["IsUnsigned"], Is.SameAs(DBNull.Value));
+                Assert.That(textRow["OID"], Is.EqualTo(25));
+
+                var numericRow = dataTypes.Rows.Cast<DataRow>().Single(r => (string)r["TypeName"] == "numeric");
+                Assert.That(numericRow["DataType"], Is.EqualTo("Decimal"));
+                Assert.That(numericRow["ProviderDbType"], Is.EqualTo((int)NpgsqlDbType.Numeric));
+                Assert.That(numericRow["IsUnsigned"], Is.False);
+                Assert.That(numericRow["OID"], Is.EqualTo(1700));
+                Assert.That(numericRow["CreateFormat"], Is.EqualTo("NUMERIC({0},{1})"));
+                Assert.That(numericRow["CreateParameters"], Is.EqualTo("precision, scale"));
+
+                var intArrayRow = dataTypes.Rows.Cast<DataRow>().Single(r => (string)r["TypeName"] == "integer[]");
+                Assert.That(intArrayRow["DataType"], Is.EqualTo("System.Int32[]"));
```

Comment:
How does test pass if you put the type instead of the name to the `DataType` column?